### DISCUSSION TITLE
Mapping a concrete source class to a Target abstract class with multiple inheritors

### DISCUSSION THREAD

**Discussion by Anto-Christen-A** [mapstruct/mapstruct#3568] [General]
I have a requirement where I'd have to map the same source class to a **target class which is abstract** and having several inheritors.
I made use of SubClassMapping and SubclassExhaustiveStrategy.RUNTIME_EXCEPTION in my Mapper but I get this error ...
"java: Subclass 'xxxxx' is already defined as a source"

I'm using Java 17 with Mapstruct 1.5.5-Final

My requirement is something like this:

```java
@SubClassMappings({
     @SubclassMapping(source = Source.class, target = TargetInheritorA.class),
     @SubclassMapping(source = Source.class, target = TargetInheritorB.class)
})
public abstract TargetClass mapToTargetClass(Source source);
```

**Comment by filiphr** [comment]
You are getting that error because you are providing the same source class for different targets. The `@SubclassMapping` works by checking if a class is an instance of something. How do you expect this to work?

What you could do is to provide a custom implementation for `mapToTargetClass` and based on your own rules delegate to the appropriate mapping methods for `TargetInheritorA` and `TargetInheritorB`
